BubblesKotlin/app/src/main/java/com/example/android/bubbles/data/Contact.kt

```diff
@@ -45,7 +46,10 @@ abstract class Contact(
     }
 
     val iconUri: Uri
-        get() = Uri.parse("content://com.example.android.bubbles/icon/$id")
+        get() = "content://com.example.android.bubbles/icon/$id".toUri()
+
+    val shortcutId: String
+        get() = "contact_$id"
```

Comment:
Nit: the id won't change right? So this could be an actual field val, rather than a `get()`

Reply:
Done.
I think I subconsciously avoided allocating just several bytes for each objects.